com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/persistence/DirectLinkingResourceStorageLoadable.java

```diff
@@ -225,12 +225,16 @@ protected void readNodeModel(final StorageAwareResource resource, final InputStr
    */
   protected void readAssociationsAdapter(final StorageAwareResource resource, final InputStream stream) throws IOException {
     DirectLinkingEObjectInputStream objIn = new DirectLinkingEObjectInputStream(stream, null);
+    InferredModelAssociator.Adapter adapter = (InferredModelAssociator.Adapter) EcoreUtil.getAdapter(resource.eAdapters(), InferredModelAssociator.Adapter.class);

```

Comment:
why did we need to move this from line 233? Could we not just assume that if size == 0 we can install the empty adapter, otherwise we continue as before?

Reply:
done